src/nvim/optionstr.c

```diff
@@ -656,8 +1643,7 @@ char *did_set_string_option(int opt_idx, char **varp, char *oldval, char *errbuf
                             int opt_flags, int *value_checked)
 {
```

Comment:
I suppose you want to pass curwin/curbuf in a future PR ? what could be done if you dont want to modify `did_set_string_option` type yet (this is a big PR after all)  is to add local variables at the start of this function like `buf = curbuf` and `win = curwin`.

Reply:
Yes.

We can rename this to `did_set_string_option_for` and add a new `did_set_string_option` as an alias which passes `curbuf`/`curwin`.